torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -122,15 +127,16 @@ def __init__(
         # (can be set to false to save communication during gradient
         # accumulation); all-reducing without reduce-scatter is disallowed
         self.reduce_scatter_grads: bool = True
+        # TODO: whether use all_reduce_grads or if isinstance(self.mesh_info, HSDPMeshInfo):
```

Comment:
current understanding: all_reduce_grads is for ``set_requires_all_reduce(True/False)`` and ``isinstance(self.mesh_info, HSDPMeshInfo)`` is from ``fully_shard(mesh=``. I am checking both to decide if we need all-reduce

Reply:
This makes sense to me! `self.all_reduce_grads` is only useful for HSDP and does not need to be touched for FSDP.